rpcs3/Emu/Cell/lv2/sys_usbd.cpp

```diff
@@ -855,10 +902,11 @@ void usb_handler_thread::disconnect_usb_device(std::shared_ptr<usb_device> dev,
 		send_message(SYS_USBD_DETACH, dev->assigned_number);
 		sys_usbd.success("USB device(VID=0x%04x, PID=0x%04x) unassigned, handled_device=0x%x", dev->device._device.idVendor, dev->device._device.idProduct, dev->assigned_number);
 		dev->assigned_number = 0;
+
+		if (update_usb_devices)
```

Comment:
why is this moved here?

Reply:
Ah yeah, i must have missed that when i was cleaning up and moved this out of its own method

Reply:
Right, i remember why i did this now
Theres a scenario where a device may have already been removed because of a failed transfer, so it was possible that it was not in `usb_devices` anymore

I've pushed a better fix for that now.